app/src/main/java/com/example/taskyy/ui/viewmodels/ReminderViewModel.kt

```diff
@@ -156,14 +158,16 @@ class ReminderViewModel @Inject constructor(
         viewModelScope.launch {
             when (val updateDb = agendaRepository.saveReminderToDB(newReminder)) {
                 is Result.Success -> {
-                    eventChannel.send(ReminderEvent.SaveSuccessful)
-                    when (val updateAPI = agendaRepository.updateReminderToApi(newReminder)) {
-                        is Result.Success -> {
+                    applicationScope.launch {
+                        eventChannel.send(ReminderEvent.SaveSuccessful)
+                        when (val updateAPI = agendaRepository.updateReminderToApi(newReminder)) {
+                            is Result.Success -> {
+                            }
+
+                            is Result.Error -> {
+                            }
                         }
-                        is Result.Error -> {
-                            agendaRepository.addFailedReminderToRetry(newReminder)
-                        }
-                    }
+                    }.join()
```

Comment:
Implementing the exact caching strategy isn't something for the VM, that's data logic. The repository should just expose a `saveReminder` function which can be called by the VM